File: packages/mobile/src/lib/logger.ts
type LogLevel = "debug" | "info" | "warn" | "error";

interface Logger {
  debug: (...args: unknown[]) => void;
  info: (...args: unknown[]) => void;
  warn: (...args: unknown[]) => void;
  error: (...args: unknown[]) => void;
}

/**
 * Create a namespaced logger.
 *
 * In production (`__DEV__ === false`), `debug` and `info` are no-ops.
 * `warn` and `error` always log.
 */
export function createLogger(namespace: string): Logger {
  const prefix = `[${namespace}]`;

  return {
    debug: __DEV__
      ? (...args: unknown[]) => console.debug(prefix, ...args)
      : () => {},

    info: __DEV__
      ? (...args: unknown[]) => console.log(prefix, ...args)
      : () => {},

    warn: (...args: unknown[]) => console.warn(prefix, ...args),

    error: (...args: unknown[]) => console.error(prefix, ...args),
  };
}

File: lib/services/logger.dart
import 'package:flutter/foundation.dart';

/// 带命名空间的日志。对应 RN 版 src/lib/logger.ts。
///
/// Release 构建里 debug / info 是空实现；warn / error 始终输出。
class Logger {
  const Logger(this.namespace);

  final String namespace;

  String get _prefix => '[$namespace]';

  void debug(Object? message) {
    if (kDebugMode) debugPrint('$_prefix $message');
  }

  void info(Object? message) {
    if (kDebugMode) debugPrint('$_prefix $message');
  }

  void warn(Object? message) {
    debugPrint('$_prefix WARN $message');
  }

  void error(Object? message) {
    debugPrint('$_prefix ERROR $message');
  }
}
